native-schema-registry/python/PyGsrSerDe/KafkaSerDe.py

```diff
@@ -0,0 +1,51 @@
+from .PyGsrSerDe import *
+from .AvroSerDe import *
+
+class GlueSchemaRegistryKafkaSerializer:
+    def __init__(self):  # data format design needs discussion
+        self.data_format = "AVRO"
+        serializers = {
+            "AVRO": AvroSerializer(),
+            "JSON": "JSONSerializer()",  # placeholders
+            "PROTOBUF": "ProtobufSerializer()",
+        }
+        self.serializer = serializers.get(self.data_format)
+
+    def serialize(self, data, schema, topic):
+        if data is None:
+            return None
+        if schema is None:
+            return None
+
+        serialized_record = self.serializer.serialize(data, schema)
+
+        gsr_schema = GlueSchemaRegistrySchema("schema_name", schema, self.data_format)
+        gsr_serializer = GlueSchemaRegistrySerializer
```

Comment:
We don't want to create a new instance of the underlying serializer for every message. This can lead to cache not getting re-used.

Reply:
Fixed